### PR TITLE
[REG-1298] Create Unity Local Bots Menu Option

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- This goes on top of REG-1288
- Adds the ability to create Unity Bots locally.  I split this off from the syncing to RG pr to keep the PRs easier to review

- There is a lot still to come in this area so I'll try to explain the future here...
  - You create the bot in unity, it gets a temporary negative bot Id until it is synched to RG.  This bot is fully workable/runnable even without syncing to RG
  - Unity Bots must live in a specific directory path in your project by convention
  - There will be a menu option to sync Unity bots for your account to/from RG.  Right now this is going to sync them ALL if their checksum doesn't match or they don't have a real Id yet, but we can add a screen for picking and choosing specific ones later.
  - The namespace for each bot MUST be unique and ideally should match your bot folder name by convention for consistency.  (they are currently generated with unique namespaces/folder-names)
    - Documentation updates will need to cover this topic

### CODE REVIEW

**Comment by vontell** [comment]
@RG-nAmKcAz it looks like you may need to rebase or merge from main? GitHub is buggy sometimes with this and you may need to make a new PR or force push the rebase/merge

File: Editor/Scripts/BotManagement/RGBotSynchronizer.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        {
+            // create a new bot folder
+            var folderName = _this.CreateBotFolder("NewRGBot", out var botId);
+
+            // create the assets
+            _this.CreateNewBotAssets(folderName, "NewRGBot", botId);
+        }
+        
+        //[MenuItem("Regression Games/Synchronize Bots with RG")]
+        //TODO (REG-1306): Implement me
```
Comment (vontell) [inline Editor/Scripts/BotManagement/RGBotSynchronizer.cs:30]: We might want to add this as an experiment - we assume they want to sync bots with RG, but do they actually?

Comment (vontell) [reply Editor/Scripts/BotManagement/RGBotSynchronizer.cs:30]: My hypothesis is that for their own bots that they share across games, and for bots that we provide as a starting point, then yes? But I don't know!

File: Runtime/Scripts/RGOverlayMenu.cs [hunk 2]
```diff
@@ -173,9 +174,18 @@ public void AddBot()
 
         public void TempTestAddLocalBot()
         {
-            // start the local bot
-            // TODO (REG-1291): Test Code: This number is meaningless until we lookup bots locally by id
-            RGBotRuntimeManager.GetInstance()?.StartBot(-123456789L);
+            // TODO (REG-1291): Temporary code to get bot 0.. this whole method will go away in
```
Comment (vontell) [inline Runtime/Scripts/RGOverlayMenu.cs:177]: This will go away in what?

**Review by vontell — APPROVED** [review]
yeah GitHub is weird...

File: Editor/Scripts/BotManagement/RGBotSynchronizer.cs (hunk 1 above)
Comment (RG-nAmKcAz) [reply Editor/Scripts/BotManagement/RGBotSynchronizer.cs:30]: ok, i already had this broken out as a separate task.  I'll move it to next cycle for further discussion as I was working on getting the overlay correct first anyway :D

Comment (RG-nAmKcAz) [reply Editor/Scripts/BotManagement/RGBotSynchronizer.cs:30]: Note added to Linear task here https://linear.app/regression-games/issue/REG-1306/sync-unity-local-bots-with-rg

File: Runtime/Scripts/RGOverlayMenu.cs (hunk 2 above)
Comment (vontell) [reply Runtime/Scripts/RGOverlayMenu.cs:177]: That's what I thought but was double checking, my brain had a hard time parsing that comment 😂

File: Runtime/Scripts/RGBotLocalRuntime/RGBotRuntimeManager.cs [hunk 3]
```diff
@@ -26,43 +36,89 @@ public void Awake()
...
         public long StartBot(long botId)
         {
-            RGBotInstance botInstance = new RGBotInstance();
-            
-            // TODO (REG-1298): Define unique botInstance Id ; for now, take the first bytes of a UUID... which isn't as unique as it sounds
-            botInstance.id = (uint) BitConverter.ToInt64(System.Guid.NewGuid().ToByteArray(), 0);
-            
+            var botInstance = new RGBotInstance
+            {
+                // TODO (REG-1298): Define unique botInstance Id ; for now, take the first bytes of a UUID... which isn't as unique as it sounds
```
Comment (abeizer) [inline Runtime/Scripts/RGBotLocalRuntime/RGBotRuntimeManager.cs:71]: Does this need a different task ID?

**Review by abeizer — APPROVED** [review]

**Review by vontell — APPROVED** [review]

**Comment by RG-nAmKcAz** [comment]
Merged as part of REG-1291

**Pull request closed without merging**